File: GenericProducerConsumer/src/Consumer.java
import java.util.Queue;

public class Consumer <MyType>{
    Queue<MyType> queue;
    public Consumer(Queue<MyType> queue) {
        this.queue = queue;
    }
    public void consume() {
        MyType item = queue.remove();
        System.out.println("Consuming item from queue: " + item);
    }
}

File: os/producer_consumer/consumer.cpp
#include <string>
#include <queue>
using namespace std;

template <typename T>
class consumer {
    private:
        queue<T>* q;
    public:
        consumer(queue<T>* q) {
            this->q = q;
        }
        T consume() {
            T item = q->front();
            q->pop();
            return item;
        }
};